### PR TITLE
AFGO implementation

### LINKED ISSUE #4: Add Aries Framework Go Docker implementation
Someone from SecureKey please help.

@troyronda please ping the appropriate folks.

### PR DESCRIPTION
**Pull request by baha-ai**
Including both VCs and Presentations in both json-ld and jwt formats.
It supports all the keys mentioned in the README.

Signed-off-by: Mykhailo Sizov <mykhailo.sizov@securekey.com>
Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>

closes #4

### CODE REVIEW

File: data/credentials/credential-3.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
       "linkRelationship": "OrganizationPresentationEndpoint"
     }
   ],
-  "issuanceDate": "2016-12-31T23:59:60Z",
```
Comment (Baha-sk) [inline data/credentials/credential-3.json:17]: 60 is an invalid seconds value

**Comment by fqutishat** [comment]
closes https://github.com/hyperledger/aries-framework-go/issues/3288

**Comment by fqutishat** [comment]
@OR13 can you review

**Review by fqutishat — APPROVED** [review]

**Review by OR13 — APPROVED** [review]

**Comment by OR13** [comment]
This is awesome!

File: data/credentials/credential-3.json (hunk 1 above)
Comment (OR13) [reply data/credentials/credential-3.json:17]: ; ) 

These are issues that the normative requirements of the next spec will need to address.

Comment (OR13) [reply data/credentials/credential-3.json:17]: I would welcome some nastier example cases, for us to address.

**Comment by fqutishat** [comment]
@Baha-sk when all test cases passing can you post screenshot.

**Comment by fqutishat** [comment]
@OR13 baha will post screenshot when all test cases are passing

**Comment by Baha-sk** [comment]
@OR13 when I run the VC verification commands for afgo as per the README doc, eg:
```
% IMPLEMENTATION=afgo
INPUT=/data/implementations/$IMPLEMENTATION/credential-0--key-0-ed25519.vc.json    
OUTPUT=/data/implementations/$IMPLEMENTATION/credential-0--key-0-ed25519.vc.verification.json 

docker-compose run $IMPLEMENTATION \
credential verify \
--input $INPUT \
--output $OUTPUT
```
or jwt :

```
% IMPLEMENTATION=afgo     
INPUT=/data/implementations/$IMPLEMENTATION/credential-0--key-0-ed25519.vc-jwt.json
OUTPUT=/data/implementations/$IMPLEMENTATION/credential-0--key-0-ed25519.vc-jwt.verification.json

docker-compose run $IMPLEMENTATION \
credential verify \
--input $INPUT \
--output $OUTPUT
```
the output file content is as expected:
```
{
    "verified": true
}
```

but when running the verify command (`npm run report:evaluate`), all vc and presentation results of afgo results fail in index.html, while the jwt ones are all successful (except maybe for one rsa key that also fails in jwt format).

I tried to run the doker-compose command using transmute with afgo credential files like so:
```
% IMPLEMENTATION=transmute
INPUT=/data/implementations/afgo/credential-0--key-0-ed25519.vc.json               
OUTPUT=/data/implementations/afgo/credential-0--key-0-ed25519.vc.verification.json               

docker-compose run $IMPLEMENTATION \
credential verify \
--input $INPUT \
--output $OUTPUT
```

but I get this output error: 
```
Creating jws-test-suite_transmute_run ... done
✨ credential [action]

Verify a verifiable credential

Options:
      --version  Show version number                                   [boolean]
  -h, --help     Show help                                             [boolean]
  -i, --input    Path to input document                               [required]
  -o, --output   Path to output document                              [required]
  -k, --key      Path to key
  -f, --format   Input format                                    [default: "vc"]

RangeError: Maximum call stack size exceeded
    at JSON.stringify (<anonymous>)
    at Error.value (/home/node/node_modules/@transmute/linked-data-proof/dist/linked-data-proof.cjs.development.js:626:23)
    at JSON.stringify (<anonymous>)
    at Error.value (/home/node/node_modules/@transmute/linked-data-proof/dist/linked-data-proof.cjs.development.js:626:23)
    at JSON.stringify (<anonymous>)
    at Error.value (/home/node/node_modules/@transmute/linked-data-proof/dist/linked-data-proof.cjs.development.js:626:23)
    at JSON.stringify (<anonymous>)
    at Error.value (/home/node/node_modules/@transmute/linked-data-proof/dist/linked-data-proof.cjs.development.js:626:23)
    at JSON.stringify (<anonymous>)
    at Error.value (/home/node/node_modules/@transmute/linked-data-proof/dist/linked-data-proof.cjs.development.js:626:23)
ERROR: 1
```

any idea how to troubleshoot this? not sure how JSON.stringify() spits the error about `this` instance, but it's probably too [large](https://stackoverflow.com/questions/14290620/maximum-call-stack-size-exceeded-during-json-stringifynavigator)?

**Comment by fqutishat** [comment]
@OR13 @Baha-sk I found the issue. Will push fix soon

**Comment by Baha-sk** [comment]
Currently, all tests are passing except for credential 2 as per @fqutishat's question to @OR13  above. Investigating next step to resolve the issue.
Screen Shot 2022-08-10 at 12 54 03 PM

**Comment by OR13** [comment]
IMO, implementations should not be mutating payload arguments, but obviously the spec isn't clear enough on this.

When in doubt, do not mutate a function parameter... ever.

**Comment by Baha-sk** [comment]
Screen Shot 2022-08-10 at 1 55 33 PM
all checks are passing now

**Comment by fqutishat** [comment]
@OR13 can merge all AFGO test is passing

**Comment by baha-ai** [comment]
great thank you @OR13, when will https://identity.foundation/JWS-Test-Suite/#implementations be updated with afgo?

**Comment by OR13** [comment]
it should happen automatically... if things were registered properly.

**Comment by baha-ai** [comment]
The reason i'm asking is that I don't see Microsoft in `index.json`, while it does show in the link above. I thought it would take some manual update somewhere to get afgo in there as well.

**Comment by baha-ai** [comment]
it seems to show now.. thanks!

**Pull request merged**